tests/event_test.py

```diff
@@ -287,3 +287,34 @@ def test_logical_and(self) -> None:
         data = json.loads(trigger.generate_subscription(logical_and=False))
         self.assertSequenceEqual(
             data['logicalAndCharactersWithWorlds'], 'false')
+        
+
+    def test_regression_66_string_event_with_world_filter(self) -> None:
+        """https://github.com/leonhard-s/auraxium/issues/66"""
+        trigger = auraxium.Trigger('ContinentLock', worlds=[1])
+        _ = json.loads(trigger.generate_subscription())
+
+    def test_logical_and_autoselect(self) -> None:
+        """Test the auto-insertion of the logicalAnd flag."""
+
+        # TODO: Support logicalAnd* auto-insertion for triggers defined using
+        # event names rather than types
+
+        event_variants = (auraxium.event.Death,)
```

Comment:
Does extending this tuple with the string variant `"Death"` still pass the test? If it does, we should do so and remove the `# TODO:` comment from line 300 to match.

Reply:
I'm not sure.  
I've never done unittesting before and having a fair bit of faff trying to get it to work :p  Just doing the test without making changes fails with `KeyError: 'logicalAndCharactersWithWorlds'` so I'm not sure if I've got it setup right.

Reply:
I believe the new version again breaks the class-based insertion, I have a suggested fix in the worlds.